---
layout: issue
title: "multiple calls with the same Zend_Rest_Client  instance not working"
id: ZF-5176
---

ZF-5176: multiple calls with the same Zend\_Rest\_Client instance not working
-----------------------------------------------------------------------------

 Issue Type: Bug Created: 2008-12-06T22:20:01.000+0000 Last Updated: 2012-03-14T05:54:20.000+0000 Status: Resolved Fix version(s): - 1.7.2 (23/Dec/08)
 
 Reporter:  Onno Schmidt (fino69)  Assignee:  Satoru Yoshida (satoruyoshida)  Tags: - Zend\_Rest\_Client
 
 Related issues: 
 Attachments: 
### Description

I found a problem using 1 instance of Zend\_Rest\_Client and doing multiple calls with parameters to the same server. The problem is that the method is never being set in the data array with the second call.

This worked fine with version 1.6.2, so it looks like the last changes in the \_\_call() method in Zend\_Rest\_Client broke the external interface.

I suggest to reset the data array after the call to the webservice is being made in the \_\_call method. Or to create a reset() method for resetting the data array.

$client = new Zend\_Rest\_Client("http://localhost/service.php");

print\_r($client->getMoo(66)->get());

returns: Zend\_Rest\_Client Object ( [\_data:protected] => Array ( [method] => getMoo [arg1] => 44 [getMoo] => 44 [rest] => 1 )

print\_r($client->getBlah(88)->get());

returns: Zend\_Rest\_Client Object ( [\_data:protected] => Array ( [method] => getMoo [arg1] => 44 [getMoo] => 44 [rest] => 1 [getBlah] => 77 )

?>

 

 

### Comments

Posted by old of Satoru Yoshida (yoshida@zend.co.jp) on 2008-12-11T02:31:33.000+0000

Solved in SVN r13151

 

 